apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

```diff
@@ -236,13 +249,89 @@ export class CheckoutComponent implements OnInit {
   }
 
   submitCheckout() {
+    // check checkout form and if terms are accepted
+    this.checkFormValidity();
+
     if (this.isLoggedIn) {
       this.createOrder();
     } else {
       this.createOrderWithNewUser();
     }
   }
 
+  checkFormValidity() {
+    if (!this.checkoutForm.valid || !this.isTermsAndConditionsAccepted) {
+      this.checkoutInvalid = true;
+      this.checkShippingInputFields();
+      if (!this.billingAddressIsTheSame) {
+        if (!this.billingAddressForm.valid) {
```

Comment:
Can be combined into
```suggestion
      if (!this.billingAddressIsTheSame && !this.billingAddressForm.valid) {
```